pkg/virt-api/api_test.go

```diff
@@ -185,128 +324,415 @@ var _ = Describe("Virt-api", func() {
 		}, 5)
 
 		It("should create apiservice endpoint if one doesn't exist", func(done Done) {
-			app.signingCertBytes = []byte("fake")
-			apiService := &apiregistrationv1beta1.APIService{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: subresourceAggregatedApiName,
-					Labels: map[string]string{
-						v1.AppLabel: "virt-api-aggregator",
-					},
-				},
-				Spec: apiregistrationv1beta1.APIServiceSpec{
-					Service: &apiregistrationv1beta1.ServiceReference{
-						Namespace: namespaceKubevirt,
-						Name:      "virt-api",
-					},
-					Group:                v1.SubresourceGroupName,
-					Version:              v1.SubresourceGroupVersion.Version,
-					CABundle:             app.signingCertBytes,
-					GroupPriorityMinimum: 1000,
-					VersionPriority:      15,
-				},
-			}
+			expectedApiService := app.subresourceApiservice()
+			expectedApiService.Kind = "APIService"
+			expectedApiService.APIVersion = "apiregistration.k8s.io/v1beta1"
 			server.AppendHandlers(
 				ghttp.CombineHandlers(
-					ghttp.VerifyRequest("GET", "/api/v1/namespaces/kubevirt/apiservices/"+subresourceAggregatedApiName),
+					ghttp.VerifyRequest("GET", "/apis/apiregistration.k8s.io/v1beta1/apiservices/"+subresourceAggregatedApiName),
 					ghttp.RespondWithJSONEncoded(http.StatusNotFound, nil),
 				),
 				ghttp.CombineHandlers(
-					ghttp.VerifyRequest("POST", "/api/v1/namespaces/kubevirt/apiservices"),
-					ghttp.RespondWithJSONEncoded(http.StatusOK, apiService),
+					ghttp.VerifyRequest("POST", "/apis/apiregistration.k8s.io/v1beta1/apiservices"),
```

Comment:
this is a different api path than what was used previously.

Reply:
Maybe setting "master" to server.URL() is very wrong, and a newbie failure.

Reply:
i see what's going on here now. we're dynamically creating a new 'aggregator client" in the createSubresourceApiservice() function.

We need a way to generate the aggregatorClient needs to live on the app object like virtCli does, and we need to be generating that client in the BeforeEach() function. That prevents you from having to set the master flag which might have unexpected consequences later on.